custom-aa/test/utils/utils.ts

```diff
@@ -0,0 +1,218 @@
+
+export class Utils {
+  private multisigAddress: any;
+  private owner1: any;
+  private owner2: any;
+  private salt: any;
+  private aaFactory: Contract | undefined;
+  private factoryAddress: any;
+  private txHash: any;
+
+  constructor() {
+    this.multisigAddress = undefined;
+    this.owner1 = undefined;
+    this.owner2 = undefined;
+    this.salt = undefined;
+    this.aaFactory = undefined;
+    this.factoryAddress = undefined;
+    this.txHash = undefined;
+  }
+
+  async deployFactory(privateKey: string = localConfig.privateKey) {
+    // Private key of the account used to deploy
+    const wallet = new Wallet(privateKey);
+    const deployer = new Deployer(hre, wallet);
+    const factoryArtifact = await deployer.loadArtifact("AAFactory");
+    const aaArtifact = await deployer.loadArtifact("TwoUserMultisig");
+
+    // Getting the bytecodeHash of the account
+    const bytecodeHash = utils.hashBytecode(aaArtifact.bytecode);
+
+    let factory = await deployer.deploy(
+      factoryArtifact,
+      [bytecodeHash],
+      undefined,
+      [
+        // Since the factory requires the code of the multisig to be available,
+        // we should pass it here as well.
+        aaArtifact.bytecode,
+      ],
+    );
+
+    console.log(`AA factory address: ${factory.address}`);
+
+    this.factoryAddress = factory.address;
+
+    return factory;
+  }
+
+  async deployMultisig(factoryAddress: string) {
+    const AA_FACTORY_ADDRESS = factoryAddress;
+
+    const provider = new Provider(localConfig.L2Network);
+    // Private key of the account used to deploy
+    const wallet = new Wallet(localConfig.privateKey).connect(provider);
+    const factoryArtifact = await hre.artifacts.readArtifact("AAFactory");
+
+    const aaFactory = new ethers.Contract(
+      AA_FACTORY_ADDRESS,
+      factoryArtifact.abi,
+      wallet,
+    );
+
+    // The two owners of the multisig
+    const owner1 = Wallet.createRandom();
+    const owner2 = Wallet.createRandom();
+
+    this.owner1 = owner1;
+    this.owner2 = owner2;
+
+    // For the simplicity of the tutorial, we will use zero hash as salt
+    const salt = ethers.constants.HashZero;
+
+    this.salt = salt;
+
+    // deploy account owned by owner1 & owner2
+    const tx = await aaFactory.deployAccount(
+      salt,
+      owner1.address,
+      owner2.address,
+      { gasLimit: 10000000 },
+    );
+    await tx.wait(1);
+
+    this.txHash = tx.hash;
+
+    // Getting the address of the deployed contract account
+    const abiCoder = new ethers.utils.AbiCoder();
+    let multisigAddress = utils.create2Address(
+      AA_FACTORY_ADDRESS,
+      await aaFactory.aaBytecodeHash(),
+      salt,
+      abiCoder.encode(["address", "address"], [owner1.address, owner2.address]),
+    );
+
+    this.multisigAddress = multisigAddress;
+
+    console.log(`Multisig account deployed on address ${multisigAddress}`);
+
+    return multisigAddress;
+  }
+
+  async executeMultiSig(sum: string = "0.008") {
```

Comment:
I still think `executeMultiSig` is too generic a name that does not really tell us what this function is actually doing: 
1. funding the multisig with the parameter passed to the function (the param name `sum` says little about how the param will be used)
2. creates and sends a transaction that is signed by the multisig

I suggest further dividing this function into these two parts and naming the functions by what they are doing, such that by reading the function name and parameter names you already know what is going to happen